drivers/sensor/bmm150/bmm150.c

```diff
@@ -309,10 +310,16 @@ static int bmm150_sample_fetch(const struct device *dev,
 	uint16_t values[BMM150_AXIS_XYZR_MAX];
 	int16_t raw_x, raw_y, raw_z;
 	uint16_t rhall;
+	int ret;
 
 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
 			chan == SENSOR_CHAN_MAGN_XYZ);
 
+	ret = pm_device_runtime_get(dev);
```

Comment:
This is not going to behave as expected. `PM_DEVICE_ACTION_SUSPEND` is putting the device into suspend mode, which per the datasheet, loses all register settings:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/8476715/2b1b2d87-2988-4a38-bd48-6302ba584762)
No configuration done by `bmm150_attr_set`, or any of the logic from `bmm150_init_chip` is going to be applied.
This is another argument in favour of my position [here](https://github.com/zephyrproject-rtos/zephyr/pull/62779#discussion_r1339310767)

Reply:
Good point! I'll try to fix that.